### DISCUSSION TITLE
Does safetensors save the model's hierarchical structure? Is it similar to ONNX?

### DISCUSSION THREAD

**Discussion by ZDragonX** [huggingface/safetensors#406] [Q&A] [answered]
If safetensors saves the model's hierarchical structure, how can one access this structure? Is it possible to read it directly like with ONNX? If the hierarchical structure of the model is not preserved, does it mean that the original model must be read from config.json?

**Comment by julien-c** [comment, marked as answer]
safetensors does not save the model's hierarchical structure, only a flat list of tensors
